consoles/console.pm

```diff
@@ -66,13 +67,15 @@ sub screen {
 
 # helper function
 sub sshCommand {
-    my ($self, $username, $host, $gui) = @_;
-
-    my $sshopts = "-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PubkeyAuthentication=no $username\@$host";
-
-    if ($gui) {
-        $sshopts = "-X $sshopts";
+    my ($self, $username, $host, $gui, $privatekey) = @_;
+    my $sshopts = "-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no";
+    $sshopts   .= " -o PubkeyAuthentication=no" unless defined $privatekey;
+    if (defined $privatekey && -e $privatekey){
```

Comment:
@coolo @mnowaksuse I'm tempted to use a job variable (say `VIRSH_SSH_PRIVATEKEY` and use that if it's set instead of expecting the private key to be passed somehow (Something I haven't figured out yet how to)